src/snaphu/_unwrap.py

```diff
@@ -363,6 +363,14 @@ def unwrap(  # type: ignore[no-untyped-def]
         if mask is not None:
             config += f"BYTEMASKFILE {tmp_mask}\n"
 
+        # To help removing tile boundaries after unwrapping
+        if ntiles != (1, 1):
+            config += "SINGLETILEREOPTIMIZE TRUE\n"
```

Comment:
I'm trying to decide how the new `single_tile_reoptimize` option should interact with the existing `regrow_conncomps` parameter.

Currently, when running in tiled mode, we have the option to re-run SNAPHU again afterwards to regrow the connected components as though by a single tile. But that seems like a waste of time if `single_tile_reoptimize` is enabled, since the connected components will already have been generated using a single tile in that case.

So allowing both `single_tile_reoptimize` and `regrow_conncomps` to be enabled simultaneously seems undesirable.

We could:

1. Automatically disable `regrow_conncomps` if `single_tile_reoptimize` is also enabled, or
2. Remove the `regrow_conncomps` option entirely and just have the `single_tile_reoptimize` option.

I'm leaning a bit more towards option 2 since the behavior will be much simpler and users would still have the option to regrow connected components manually using [`grow_conncomps()`](https://github.com/isce-framework/snaphu-py/blob/02a091ffa1711d6e656505243929e48347173e6c/src/snaphu/_conncomp.py#L145) if they want.

@mirzaees @scottstanie what are your thoughts?

Reply:
I am confused by the SNAPHU documentation-
I can't tell if `SINGLETILEREOPTIMIZE` turned on means the steps are 

1. split tiles, unwrap each tile
2. solve a secondary problem to try to align the tiles
3. redo the final result as though you passed it on the command line like `snaphu -u ./my_already_unwrapped_result.unw ...`

or if it's just steps (1) and (3)?

If it's 1/2/3, that seems like it should be off by default, but I haven't gotten a chance to see exactly what problems Sara was seeing

Reply:
It's all three steps -- (1), (2), and (3). Look for the `onetilereopt` parameter in the `Unwrap()` function [here](https://github.com/gmgunter/snaphu/blob/5981e65f3c267906b32dd5411bc27e0c4844f216/src/snaphu.c#L128).

Reply:
If the `single_tile_reoptimize` is going to be an optional config, then I think option 1 is better to have the `regrow_conncomps ` functionality here for when we possibly need to use it
thank you @gmgunter, please feel free to take over

Reply:
Option 1 works for me.

I chatted with @scottstanie offline a bit about whether or not `single_tile_reoptimize` should be enabled by default. I think we settled on enabling it by default for now, but might revisit that later on after more testing.